modules/gui.py

```diff
@@ -0,0 +1,210 @@
+        self.display_week_list()
+
+    def show_next_week(self):
+        self.current_date += datetime.timedelta(days=7)
+        self.display_week_list()
+
+    def show_current_week(self):
+        self.current_date = datetime.date.today()
+        self.display_week_list()
+
+    def display_week_list(self):
+        self.days_listbox.delete(0, tk.END)
+
+        start, end = self.get_week_range(self.current_date)
+        week_key = f"{self.get_date_string(start)} - {self.get_date_string(end)}"
+        self.week_label.config(text=week_key)
+
+        if (week_key := f"{self.get_date_string(start)} - {self.get_date_string(end)}") in self.data:
+            week_data = self.data[week_key]
+        else:
+            week_data = {}
+
+        current_week_dates = [start + datetime.timedelta(days=i) for i in range(5)]
+        current_week_dates_strings = [self.get_date_string(date) for date in current_week_dates]
+
+        current_week_dates_strings = list(dict.fromkeys(current_week_dates_strings))
+
+        for date in current_week_dates_strings:
+            self.days_listbox.insert(tk.END, date)
+
+    def on_day_select(self, event):
+        selection = self.days_listbox.curselection()
+        if selection:
+            date_key = self.days_listbox.get(selection[0])
+            week_key = self.week_label.cget("text")
+
+            self.entry_text.delete("1.0", tk.END)
+            if week_key in self.data and date_key in self.data[week_key]:
+                entries = self.data[week_key][date_key]
+                for entry in entries:
+                    self.entry_text.insert(tk.END, f"{entry}\n")
+            self.current_day = datetime.datetime.strptime(date_key, "%d.%m.%Y").date()
+
+    def get_week_range(self, date):
+        start = date - datetime.timedelta(days=date.weekday())
+        end = start + datetime.timedelta(days=4)
+        return start, end
+
+    def get_date_string(self, date):
+        return date.strftime("%d.%m.%Y")
+
+    def process_entry(self, entry):
+        processed_entries = []
+        for line in entry.split('\n'):
+            if line.strip().startswith("§i"):
+                processed_entries.append(line.strip().replace("§i", "Illness: "))
+            elif line.strip().startswith("§v"):
+                processed_entries.append(line.strip().replace("§v", "Vacation: "))
+            else:
+                processed_entries.append(line.strip())
+        return processed_entries
+
+    def ensure_week_structure(self, date):
+        start, end = self.get_week_range(date)
+        week_key = f"{self.get_date_string(start)} - {self.get_date_string(end)}"
+        if week_key not in self.data:
+            self.data[week_key] = {}
+
+    def start_hotkey_listener(self):
+        def on_ctrl_s():
+            self.submit_entry()
+
+        keyboard.add_hotkey("ctrl+s", on_ctrl_s)
+        threading.Thread(target=keyboard.wait, args=("ctrl+s",), daemon=True).start()
+
+def run_gui():
```

Comment:
I did change the main() function name to run_gui(). The gui is not the main feature or start point of the app so it should be called through setup.py